Microsoft.Toolkit.Uwp.UI.Controls/FrostedGlass/FrostedGlass.cs

```diff
@@ -0,0 +1,181 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Graphics.Canvas.Effects;
+using Windows.UI;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Markup;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// The <see cref="FrostedGlass"/> control allows user to create layout control that enables tint and blur
+    /// </summary>
+    [TemplatePart(Name = PartGlassHost, Type = typeof(Border))]
+    [ContentProperty(Name = "Content")]
+    public partial class FrostedGlass : ContentControl
+    {
+        private const string PartGlassHost = "GlassHostElement";
+
+        private Compositor _compositor;
+        private CompositionBrush _backdropBrush;
+        private GaussianBlurEffect _glassEffect;
+        private SpriteVisual _glassVisual;
+        private Border _border;
+
+        private bool _loaded = false;
+
+        public FrostedGlass()
+        {
+            DefaultStyleKey = typeof(FrostedGlass);
+        }
+
+        /// <summary>
+        /// Update the visual state of the control when its template is changed.
+        /// </summary>
+        protected override void OnApplyTemplate()
+        {
+            if (!IsSupported)
+            {
+                return;
+            }
+
+            _border = GetTemplateChild(PartGlassHost) as Border;
+
+            if (_border != null && !_loaded)
+            {
+                CreateEffect();
+                _loaded = true;
+            }
+
+            base.OnApplyTemplate();
+        }
+
+        private void CreateEffect()
+        {
+            _border = GetTemplateChild(PartGlassHost) as Border;
```

Comment:
why not just passing the _border variable from OnApplyTemplate instead of calling this line twice which is redundant
`_border = GetTemplateChild(PartGlassHost) as Border;`